### ISSUE TITLE
F1255-16 missing some (almost all) controls

### ISSUE THREAD

**Issue by roccolord** [jaroschek/home-assistant-myuplink#48]
Hi, first off thanks for your hard work and very nice integration to myUplink!

I have a F1255-16 with the upgraded firmware to be compatible with myUplink instead of Nibe Uplink.

Everything looks to be working fine, except the controls for everything but temporary lux hot water. I have the myUplink Premium Manage package, is it so that this is not supported yet?

Is there anything I can do to help you adding support for this? I would like to be able so change temperature in the system and scheduling for instance. holiday mode and so on :)

![nibe_settings](https://github.com/jaroschek/home-assistant-myuplink/assets/164939/c42b6c39-325c-49e8-9bb5-23d5d1c7588c)

**Comment by Matmarknaden** [comment]
Hello, I share this problem. Appreciate som input / guidance (if there is a solution to this). I am also trying to get support from MyUplink (check if there are bugs on their side)

**Comment by pszewello** [comment]
As most of old users of uplink have to move to myuplink i expect to have same functionality in the api, right now i miss the climate entity, water heater entity and status of the extra heater… i will add the points from the api tomorrow here so u can take a look :)

**Comment by pszewello** [comment]
[myUplinkPointsAnswe.json](https://github.com/jaroschek/home-assistant-myuplink/files/13828970/myUplinkPointsAnswe.json)
[myUplinkDevicesAnswer.json](https://github.com/jaroschek/home-assistant-myuplink/files/13828971/myUplinkDevicesAnswer.json)
Please find the API json responses for devices and points...

**Comment by jaroschek** [comment]
> As most of old users of uplink have to move to myuplink i expect to have same functionality in the api, right now i miss the climate entity, water heater entity and status of the extra heater… i will add the points from the api tomorrow here so u can take a look :)

Hello @pszewello, thanks for providing api responses for your device and parameter points. 

I have developed the integration for monitoring of my heating in the first step. So the main functionality of the integration was to create (basic) sensor entities. Further on @7RST1 added the possibility to alter writable parameter points and allowed to control certain values through the API. 

If parameter points are writable, you should be able to change them through HA.

There is also a first implementation for a water header available, but this is currently hardcoded/limit to a device name 18760NE and specific parameter ids. Currently there is no climate entity implemented. 

As far as I understand, water heater entities and climate entities are working on combinations of different parameter points. And because myUplink provides many different parameters from many different devices of many different manufacturers it will be challenging to create entity implementations to combine the different individual parameter points from the different devices and manufacturers.

But anybody is free to provide necessary changes through an PR to add these.

**Comment by elden1337** [comment]
am in touch with a guy at Nibe regarding some parameters. Missing the heating offset for instance (if spa is active it's there but not writable, if spa is inactive it's not there at all). in the swagger that is, so not an error in this integration. Nibe F730

**Comment by pszewello** [comment]
How do you get in touch with nibe guys?

**Comment by elden1337** [comment]
> How do you get in touch with nibe guys?
Was contacted a couple of years ago when i started developing PeaqHvac which does things their own logic does not, plus started debating swedish climate actions online. I hope they answer, I'm just testing it out as we speak and yes there are differences from the old uplink.

**Comment by Mortenjoe** [comment]
The list of parameter ID's when querying /v2/devices/{deviceId}/points does not seem to be complete. I can control many parameters from the MyUplink webpage but it's only the lux hot water control that is listed as writable (VVM310 + F2120-16). Would it be possible to integrate custom parameter ID's not included in the list?

Morten

**Comment by elden1337** [comment]
There is a clear diff in what's provided in the app (and old api) vs new api yes. 

Some findings i have done:

-Electrical addon current power
-Heating offset (not the spa one but the one you can set)
-dm compressor start
-raw ventilation speed (the one in %). 

There ate probably more. These are the ones i've reported to my contact at Nibe as of now and awaiting reply.

**Comment by Mortenjoe** [comment]
Sorry for not being more clear. I have endpoints which are working on the swagger but are not listed when querying /v2/devices/{deviceId}/points. You can get the endpoints by looking at the header of the http request when using the MyUplink webpage. 

Morten

**Comment by elden1337** [comment]
Yes, like 47011 for instance. I have a dialogue with Nibe about the missing params and hope it will be resolved shortly.

**Comment by zissou1** [comment]
I have just moved to myUplink aswell and noticed some API-points missing from the old API. Have you gotten any answer from Nibe @elden1337 ?

**Comment by elden1337** [comment]
I dont have any resolvement yet no, sadly.

**Comment by jaroschek** [comment]
May be the `internalapi` is the place to look into: https://internalapi.myuplink.com/swagger/index.html

There are several APIs available through the select in the upper right corner:
![image](https://github.com/jaroschek/home-assistant-myuplink/assets/470290/eec95966-9251-4f3d-aa65-1a1af1951f6b)
 
But as the name suggests, these APIs are meant for "internal" use. Theoretically it could be used as soon as we find a way to authorize a user. But unfortunately the application credentials generated at https://dev.myuplink.com are only suitable for the public api.

May be it is somehow possible to authorizes using the `username` and `password`, like if you log into the dashboard?

**Comment by elden1337** [comment]
Awaiting response from Nibe. 47001 is set-able but not visible on get. And i've gotten info that for instance the amount of elec addon usage has been removed on purpose, so that one is no longer available with myuplink+nibe

**Comment by zissou1** [comment]
Sad about 47001, did they mention why it was removed?

**Comment by elden1337** [comment]
They have now told me it looks like a bug. So awaiting reply on when it's fixed.

**Comment by elden1337** [comment]
They have admitted to some kind of bad implementation and told me it will take a while, affecting both fm and the api side. 
This was almost 1 month ago.

**Comment by elden1337** [comment]
@LarsFransson65 I think since they said there was some issue they needed to attend to, that they are redoing part of the implementation. So let's see what params appear once ready. It's been a while now but they said it would take a while also.

**Comment by elden1337** [comment]
Not expecting anything til after May. Got notice that there are other higher prios first.
